crates/core/src/constraint_system/prove.rs

```diff
@@ -511,6 +522,32 @@ where
 	Ok(proof)
 }
 
+fn emit_max_rss() {
```

Comment:
Please move to a util module. `binius_utils::perf` or something. Also, this needs a docstring.

Regarding naming, it seems like it should be `log_max_rss` instead of `edit`, since it's logging a tracing event?

Reply:
It's actually **e**mit, as in emitting a tracing event, but I don't care about the name that much.

Now, regarding moving to the utils module. That would add `tracing` dep to the utils module. Are we alright with that?

Reply:
Oops, I misread. Naming is fine.

Regarding location, I'm happy with either a `binius_core::util` module or a `tracing` Cargo feature in the `binius_utils` crate.

Reply:
We can actually place it in `tracing-profile` as this may be handy in `ic1` as well